### PR TITLE
Source Freshdesk: add support for incremental sync #1388

### LINKED ISSUE #1388: Source Freshdesk: add support for incremental sync
## Tell us about the problem you're trying to solve
Follow up to #1387 

There are a few parts to this: 
1. Identify which streams can sync incrementally, and which bookmark you'll be using
2. Update the catalog generated in `discover` to declare which streams sync incrementally
3. Implement the incremental sync process

To incrementally sync a child entity child entity (e.g: `/customers/{customer_id}/customer_events` ), we will probably need to do a "full read" of the parent entity (get all parents), then look for the child entities that have changed since the last time we read the child entities. 
For example: if we're syncing `/customers/{customer_id}/customer_events`  then we should read all customers, then for each customer, read all the events we haven't read since the last time we performed a sync. 
It's probably not practical to keep track of the last event we synced for _each_  customer, so it may make sense to keep track of a global cursor for all customer_events. 
For example, we may just have to remember that the largest `customer_event.updated_at` value we extracted last time a sync ran was `october 10`. In which case the next sync we read all customers, then for each customer extract all the events where `updated_at >= october 10`. 

Note (from @eugene-kulak ) I am not sure how the plan above is relevant, because I haven't found nested streams

Streams (needs to be verified, Stitch.io says they are, but API is not clear about it) that support incremental sync:
- Agents
- Companies
- Conversations
- Groups
- Roles
- Satisfaction Ratings
- Tickets
- Time Entries

### PR DESCRIPTION
**Pull request by eugene-kulak**
## What
#1388

### CODE REVIEW

File: docs/integrations/sources/freshdesk.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ If there are more endpoints you'd like Airbyte to support, please [create an iss
 | Feature | Supported? |
 | :--- | :--- |
 | Full Refresh Sync | Yes |
-| Incremental Sync | Coming soon |
+| Incremental Sync | Yes (except Conversations) |
 | Replicate Incremental Deletes | Coming soon |
```
Comment (sherifnada) [inline docs/integrations/sources/freshdesk.md:31]: The API has no concept of this, right? so we shouldn't really mention it in the docs
```suggestion
```

File: airbyte-integrations/connectors/source-freshdesk/sample_files/configured_catalog.json [hunk 2]
```diff
@@ -548,11 +526,11 @@
             }
           }
         },
-        "supported_sync_modes": ["full_refresh"],
-        "source_defined_cursor": false,
+        "supported_sync_modes": ["full_refresh", "incremental"],
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-freshdesk/sample_files/configured_catalog.json:529]: does this source support full refresh if the sync mode = full refresh? or does it always look at state ?

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-freshdesk/sample_files/configured_catalog.json (hunk 2 above)
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-freshdesk/sample_files/configured_catalog.json]: unfortunately not, I have an idea to force this in the base source, but I think it makes sense to do this in a separate PR

**Pull request merged**